src/amchart/CommonSerial.js

```diff
@@ -210,135 +274,216 @@
-        this._chart.categoryAxis.fontSize = xAxis.axisFontSize();
-        this._chart.categoryAxis.color = xAxis.axisFontColor();
-        this._chart.categoryAxis.titleColor = xAxis.axisTitleFontColor();
-        this._chart.categoryAxis.titleFontSize = xAxis.axisTitleFontSize();
+        this._chart.categoryAxis.position = xAxis.position() ? xAxis.position() : "bottom";
+        this._chart.categoryAxis.autoGridCount = (this.xAxisAutoGridCount() && this.backwardsCompatible()) ? this.xAxisAutoGridCount() : xAxis.axisAutoGridCount();
+        this._chart.categoryAxis.gridPosition = (this.xAxisGridPosition() && this.backwardsCompatible())  ? this.xAxisGridPosition() : xAxis.axisGridPosition();
+        this._chart.categoryAxis.axisAlpha = (this.axisAlpha() && this.backwardsCompatible()) ? this.axisAlpha() : xAxis.axisAlpha();
+        this._chart.categoryAxis.gridAlpha = (this.xAxisGridAlpha() && this.backwardsCompatible()) ? this.xAxisGridAlpha() : xAxis.axisGridAlpha();
+        this._chart.categoryAxis.startOnAxis = (this.startOnAxis() && this.backwardsCompatible()) ? this.startOnAxis() : xAxis.startOnAxis();
+        this._chart.categoryAxis.labelRotation = (this.xAxisLabelRotation() && this.backwardsCompatible()) ? this.xAxisLabelRotation() : xAxis.axisLabelRotation();
+        this._chart.categoryAxis.title = (this.xAxisTitle() && this.backwardsCompatible()) ? this.xAxisTitle() : xAxis.axisTitle();
+
+        this._chart.categoryAxis.axisColor = (this.xAxisBaselineColor() && this.backwardsCompatible()) ? this.xAxisBaselineColor() : xAxis.axisBaselineColor();
+        this._chart.categoryAxis.axisThickness = (this.axisLineWidth() && this.backwardsCompatible()) ? this.axisLineWidth() : xAxis.axisLineWidth();
+        this._chart.categoryAxis.boldPeriodBeginning = (this.xAxisBoldPeriodBeginning() && this.backwardsCompatible()) ? this.xAxisBoldPeriodBeginning() : xAxis.axisBoldPeriodBeginning();
+        this._chart.categoryAxis.dashLength = (this.xAxisDashLength() && this.backwardsCompatible()) ? this.xAxisDashLength() : xAxis.axisDashLength();
+        this._chart.categoryAxis.fillAlpha = (this.xAxisFillAlpha() && this.backwardsCompatible()) ? this.xAxisFillAlpha() : xAxis.axisFillAlpha();
+        this._chart.categoryAxis.fillColor = (this.xAxisFillColor() && this.backwardsCompatible()) ? this.xAxisFillColor() : xAxis.axisFillColor();
+        this._chart.categoryAxis.fontSize = (this.axisFontSize() && this.backwardsCompatible()) ? this.axisFontSize() : xAxis.axisFontSize();
+        this._chart.categoryAxis.color = (this.xAxisFontColor() && this.backwardsCompatible()) ? this.xAxisFontColor() : xAxis.axisFontColor();
+        this._chart.categoryAxis.titleColor = (this.xAxisTitleFontColor() && this.backwardsCompatible()) ? this.xAxisTitleFontColor() : xAxis.axisTitleFontColor();
+        this._chart.categoryAxis.titleFontSize = (this.xAxisTitleFontSize() && this.backwardsCompatible()) ? this.xAxisTitleFontSize() : xAxis.axisTitleFontSize();
```

Comment:
not going to happen ... might as well just revert to the old code base then and revove the idea of the axis object